### PR TITLE
refactor: split Router and Http basics

### PR DESCRIPTION
**Pull request by aarcex3**
# Summary
Moved the Router class and related files/classes from `Tempest\Http`  to its own package, namely `Tempest\Routing`.

# Related Issue
#390

### CODE REVIEW

File: src/Tempest/Http/src/Commands/RoutesCommand.php [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 use Tempest\Console\Console;
 use Tempest\Console\ConsoleCommand;
-use Tempest\Http\RouteConfig;
+use Tempest\Routing\RouteConfig;
 
 final readonly class RoutesCommand
```
Comment (aidan-casey) [inline src/Tempest/Http/src/Commands/RoutesCommand.php:11]: This routes command should probably move to the router.

File: src/Tempest/View/composer.json [hunk 2]
```diff
@@ -7,11 +7,12 @@
         "masterminds/html5": "^2.9",
         "tempest/core": "self.version",
         "tempest/container": "self.version",
-        "tempest/validation": "self.version"
+        "tempest/validation": "self.version",
+        "tempest/routing": "self.version"
```
Comment (brendt) [inline src/Tempest/View/composer.json:11]: Why does view need a dependency on routing?

**Review by brendt — CHANGES_REQUESTED** [review]

File: src/Tempest/View/composer.json (hunk 2 above)
Comment (aarcex3) [reply src/Tempest/View/composer.json:11]: My bad, I'll remove it

File: src/Tempest/Routing/phpunit.xml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<phpunit xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://schema.phpunit.de/11.4/phpunit.xsd" bootstrap="vendor/autoload.php" executionOrder="depends,defects" beStrictAboutOutputDuringTests="true" displayDetailsOnPhpunitDeprecations="true" failOnPhpunitDeprecation="false" failOnRisky="true" failOnWarning="true">
+    <testsuites>
+        <testsuite name="Tempest Http">
```
Comment (aidan-casey) [inline src/Tempest/Routing/phpunit.xml:4]: Can you update this to be `Tempest Routing`?

**Review by aidan-casey — CHANGES_REQUESTED** [review]
@aarcex3 - Thanks for your work on this! Just a couple comments in the review.

I think generally the goal here is to separate the things that might be used across multiple http-related packages (e.g., `Status` enum, `ContentType` enum, `Method` enum, etc.) to the `Http` namespace so it can be used by both `HttpClient` and `Routing`. Just keep that in mind as you go about the work here. 🙂

File: src/Tempest/Routing/composer.json
[the remark this answers is not in the document]
Comment (aarcex3) [reply src/Tempest/Routing/composer.json:1]: Like a README.md?

[the remark this answers is not in the document]
Comment (aidan-casey) [reply src/Tempest/Routing/composer.json:1]: Just a short [description](https://getcomposer.org/doc/04-schema.md#description) key in the `composer.json` file. 🙂

**Comment by brendt** [comment]
@aidan-casey do you wanna followup on this?

**Comment by brendt** [comment]
I'm going to close it for now due to inactivity and merge conflicts.

**Pull request closed without merging**